src/esperanza/walletextension.cpp

```diff
@@ -286,6 +286,12 @@ bool WalletExtension::SendDeposit(const CKeyID &keyID, CAmount amount,
   const FinalizationState *fin_state = m_dependencies.GetFinalizationStateRepository().GetTipState();
   assert(fin_state);
 
+  const esperanza::Result is_valid = fin_state->ValidateDeposit(keyID, amount);
```

Comment:
Sounds reasonable to validate deposit before adding it to the wallet.
WDYS if we add validation for other finalizer commits? Because it seems the same issue can happen with other types of txs.

Also, maybe `ContextualCheck*` would be more appropriate?

Reply:
In this PR I won't use `ContextualCheck` because it leads to double check of any spawned commit. I'd keep simple approach that covers most annoying cases for deposit for now.

I'm thinking how to rework commits spawning with single ContextualCheck that would be performed once and prior adding transaction to the wallet, but it will go to the separate PR.

Reply:
There is a "correction" comment to this issue https://github.com/dtr-org/unit-e/issues/996#issuecomment-483694251 that the problem is not only with the deposit but with all finalizer commits. I think we then shouldn't resolve that issue once we merge this PR.